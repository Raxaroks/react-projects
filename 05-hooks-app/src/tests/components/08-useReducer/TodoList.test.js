import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import TodoList from '../../../components/08-useReducer/TodoList';
import { demoTodos } from '../../fixtures/demoTodos';


describe('Pruebas en el componente TodoListItem', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const componentWrapper = shallow(
        <TodoList
            todos={ demoTodos }
            handleDelete={ handleDelete }
            handleToggle={ handleToggle }
        />
    );
   
    test('Debe de renderizarse correctamente.', () => {
        expect(componentWrapper).toMatchSnapshot();
    });  
    
    test('Debe de tener una cantidad de 2 TodoListItem.', () => {
        expect(componentWrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect( componentWrapper.find('TodoListItem').at(0).prop('handleDelete') ).toEqual( expect.any(Function) );
    });
    
});
