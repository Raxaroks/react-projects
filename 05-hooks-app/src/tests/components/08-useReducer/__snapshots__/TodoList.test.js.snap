// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente TodoListItem Debe de renderizarse correctamente. 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender ReactJS",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender MongoDB",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
